tests/array/array-compare.dora

```diff
@@ -0,0 +1,18 @@
+fun main() {
+  let a = Array[Int32](1, 2, 3, 3, 4, 5, 4, 5, 4, 3, 3);
+  let b = Array[Int32](1, 2, 3, 3, 4, 5, 4, 5, 4, 3, 4);
+  let c = Array[Int32](1, 2, 3, 3, 4, 5, 4, 5, 4, 3);
+  let d = Array[Int32](2, 3, 3, 4, 5, 4, 5, 4, 3, 4);
+  let e = Array[Int32]();
+
+  assert(Array[Int32]::compare(a, 0L, a, 0L, 11L));
+  assert(Array[Int32]::compare(a, 0L, a, 0L, 12L).not()); // desired?
```

Comment:
shouldn't this be fail because of out-of-bounds access?

Reply:
I haven't thought thoroughly whether `len` should be "exactly `len` elements" or "up to `len` elements", so I opted for the more "lenient" interpretation, with the plan to revisit this when I come around to write documentation for this method.

Reply:
I added  a todo documenting this.